core/storage/user/gae_models.py

```diff
@@ -2278,6 +2265,9 @@ class PendingDeletionRequestModel(base_models.BaseModel):
 
     # The email of the user.
     email = ndb.StringProperty(required=True)
+    # Role of the user. Needed to decide which storage models have to be deleted
+    # for it.
+    role = ndb.StringProperty(required=True)
```

Comment:
Are you sure about this? Could it change in the interim (through the admin page)?

Reply:
Should be fine. When `PendingDeletionRequestModel` is created `UserSettingsModel` is marked as deleted and there is no way to change the role.

Reply:
@vojtechjelinek thanks for responding on this.
@seanlip I have checked it too, once `UserSettingsModel` is marked as deleted, the `AdminRoleHandler` will not be able to set the new role for the user using `user_services.update_user_role` - a "User not found" error will be raised because when `deleted=True` for a `UserSettingsModel` the corresponding user entry is not fetched in `user_services.get_user_settings`.